Description: Replace orderedhash gem with active_support
 Gem orderedhash <https://rubygems.org/gems/orderedhash/>
 has several problems:
 * not packaged in Debian (so not used by some other software)
 * does not look maintained (last version 0.0.6 is from 2008)
 * no license file included (just one mention of public domain
   in a source file)
 .
 On the other hand, replacement active_support gem is rather
 popular (albeit somewhat big) and does not experience any of
 these problems.
Author: Valentin Vidic <Valentin.Vidic@CARNet.hr>
Last-Update: 2017-06-17
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/pcsd/config.rb
+++ b/pcsd/config.rb
@@ -1,5 +1,5 @@
 require 'json'
-require 'orderedhash'
+require 'active_support/ordered_hash'
 
 require 'cluster.rb'
 require 'permissions.rb'
@@ -124,15 +124,15 @@
   end
 
   def text()
-    out_hash = OrderedHash.new
+    out_hash = ActiveSupport::OrderedHash.new
     out_hash['format_version'] = CURRENT_FORMAT
     out_hash['data_version'] = @data_version
     out_hash['clusters'] = []
-    out_hash['permissions'] = OrderedHash.new
+    out_hash['permissions'] = ActiveSupport::OrderedHash.new
     out_hash['permissions']['local_cluster'] = []
 
     @clusters.each { |c|
-      c_hash = OrderedHash.new
+      c_hash = ActiveSupport::OrderedHash.new
       c_hash['name'] = c.name
       c_hash['nodes'] = c.nodes.uniq.sort
       out_hash['clusters'] << c_hash
@@ -226,10 +226,10 @@
   end
 
   def text()
-    tokens_hash = OrderedHash.new
+    tokens_hash = ActiveSupport::OrderedHash.new
     @tokens.keys.sort.each { |key| tokens_hash[key] = @tokens[key] }
 
-    out_hash = OrderedHash.new
+    out_hash = ActiveSupport::OrderedHash.new
     out_hash['format_version'] = CURRENT_FORMAT
     out_hash['data_version'] = @data_version
     out_hash['tokens'] = tokens_hash
--- a/pcsd/pcsd-cli.rb
+++ b/pcsd/pcsd-cli.rb
@@ -4,14 +4,14 @@
 require 'etc'
 require 'json'
 require 'stringio'
-require 'orderedhash'
+require 'active_support/ordered_hash'
 
 require 'bootstrap.rb'
 require 'pcs.rb'
 require 'auth.rb'
 
 def cli_format_response(status, text=nil, data=nil)
-  response = OrderedHash.new
+  response = ActiveSupport::OrderedHash.new
   response['status'] = status
   response['text'] = text if text
   response['data'] = data if data
--- a/pcsd/permissions.rb
+++ b/pcsd/permissions.rb
@@ -1,4 +1,4 @@
-require 'orderedhash'
+require 'active_support/ordered_hash'
 
 module Permissions
 
@@ -104,7 +104,7 @@
     end
 
     def to_hash()
-      perm_hash = OrderedHash.new
+      perm_hash = ActiveSupport::OrderedHash.new
       perm_hash['type'] = @type
       perm_hash['name'] = @name
       perm_hash['allow'] = @allow_list.uniq.sort
